framework/users.py

```diff
@@ -0,0 +1,189 @@
+        set explicity, this will return `None`.
+        """
+        return self.__user_id
+
+
+    def auth_domain(self):
+        """Obtains the user's authentication domain.
+
+        Returns:
+        A string containing the authentication domain. This method is internal and
+        should not be used by client applications.
+        """
+        return self.__auth_domain
+
+
+    def federated_identity(self):
+        """Decommissioned, don't use.
+
+        Returns:
+        A string containing the federated identity of the user. If the user is not
+        a federated user, `None` is returned.
+        """
+        return self.__federated_identity
+
+
+    def federated_provider(self):
+        """Decommissioned, don't use.
+
+        Returns:
+        A string containing the federated provider. If the user is not a federated
+        user, `None` is returned.
+        """
+        return self.__federated_provider
+
+
+    def __unicode__(self):
+        return six_subset.text_type(self.nickname())
+
+    def __str__(self):
+        return str(self.nickname())
+
+    def __repr__(self):
+        values = []
+        if self.__email:
+            values.append("email='%s'" % self.__email)
+        if self.__federated_identity:
+            values.append("federated_identity='%s'" % self.__federated_identity)
+        if self.__user_id:
+            values.append("_user_id='%s'" % self.__user_id)
+        return 'users.User(%s)' % ','.join(values)
+
+    def __hash__(self):
+        if self.__federated_identity:
+            return hash((self.__federated_identity, self.__auth_domain))
+        else:
+            return hash((self.__email, self.__auth_domain))
+
+    def __cmp__(self, other):
+        if not isinstance(other, User):
+            return NotImplemented
+        if self.__federated_identity:
+            return cmp((self.__federated_identity, self.__auth_domain),
+                        (other.__federated_identity, other.__auth_domain))
+        else:
+            return cmp((self.__email, self.__auth_domain),
+                        (other.__email, other.__auth_domain))
+
+    def get_current_user():
+        """Retrieves information associated with the user that is making a request.
+
+        Returns:
+
+        """
+        try:
+            return User()
+        except UserNotFoundError:
+            return None
+
+
+    def is_current_user_admin():
+        """Specifies whether the user making a request is an application admin.
+
+        Because administrator status is not persisted in the datastore,
+        `is_current_user_admin()` is a separate function rather than a member function
+        of the `User` class. The status only exists for the user making the current
+        request.
+
+        Returns:
+        `True` if the user is an administrator; all other user types return `False`.
+        """
+        return (os.environ.get('USER_IS_ADMIN', '0')) == '1'
+
+
+def get_current_user():
```

Comment:
This also looks pretty good.  However, we could not deploy this to the production server in a single step.  Existing users who have a browser tab open on our site and information typed into a form before the deployment would get an error if they submit the form after the deployment.   We need to check if the user is signed in with the old GAE users library.  I think that case should be supported for POST requests only.  If we continued to honor old GAE users sessions for GETs, then users would never realize that they need to sign in using the new method.

Reply:
Okay... Just to confirm that I understood correctly... We will allow POSTS requests sent by users logged in using the old GAE users library but for GET requests, they will have to login using the newer Google Sign-In

Reply:
@jrobbins, I have implemented it. See `framework/basehandlers.py: line 75` and `framework/permissions.py: Line 33`. Please review.

Reply:
Your testing procedure sounds right to me.

The query string parameter to open a dialog also seems like a good solution.  Thanks for thinking through that.  It would be better to initiate the log-in process for the user rather than call alert(), but that can be improved in a later CL.